src/client/dfuse/pil4dfs/int_dfs.c

```diff
@@ -5044,12 +5004,12 @@ update_cwd(void)
 	cwd = get_current_dir_name();
 
 	if (cwd == NULL) {
-		D_ERROR("failed to get CWD with get_current_dir_name(). Quit.");
+		D_ERROR("failed to get CWD with get_current_dir_name(). Quit.\n");
```

Comment:
This shouldn't have `Quit.` in the string.
```suggestion
		D_ERROR("failed to get CWD with get_current_dir_name().\n");
```

Assuming you do need to exit here and I think it's likely you do if I've read the code correctly then D_CRIT or D_FATAL are probably better here.  The closest macro we have is probably D_ASSERT() however that's used for internal consistency failures whereas I think this is more a case of some operation isn't supported.

Adding a new macro to the head of this file to handle these cannot-continue cases would be helpful.  It probably wants to call abort() rather than exit(1) as well.

Reply:
D_FATAL() is used to replace D_ERROR(). abort() is used to replace exit(). Thank you!